script.js

```diff
@@ -1,4 +1,193 @@
+  var final = "";
+  var array = getCurrentPlayer();
+  if (currentMode == MODE_LOWEST_WINS) {
+    //generate lowest number possible from array
+    array.sort((a, b) => a - b);
+  } else {
+    //geenerate highest number possible from array
+    array.sort((a, b) => b - a);
+  }
+  for (var i = 0; i < array.length; i += 1) {
+    final += `${array[i]}`;
+  }
+  if (currentPlayer == 1) {
+    playerOneFinal = final;
+  } else {
+    playerTwoFinal = final;
+  }
+  return final;
+};
+
+var calculateWinner = function () {
+  var output = "";
+  if (
+    (playerOneFinal > playerTwoFinal && currentMode == MODE_NORMAL) ||
+    (playerOneFinal < playerTwoFinal && currentMode == MODE_LOWEST_WINS)
+  ) {
+    output += `Player 1 Wins! Player 1: ${playerOneFinal} vs Player 2: ${playerTwoFinal}`;
+    playerOneScore += 1;
+  } else if (
+    (playerOneFinal < playerTwoFinal && currentMode == MODE_NORMAL) ||
+    (playerOneFinal < playerTwoFinal && currentMode == MODE_LOWEST_WINS)
+  ) {
+    output += `Player 2 Wins! Player 1: ${playerOneFinal} vs Player 2: ${playerTwoFinal}`;
+    playerTwoScore += 1;
+  } else output += `It's a draw`;
+
+  return output;
+};
+
+var getPlayerCombinedResult = function (order) {
+  var total = updatePlayerTotal(order);
+  var output = `Player ${currentPlayer}, you chose Dice ${order} first.
+  <br>Your number is ${total}`;
+  if (currentPlayer == 1) {
+    currentPlayer += 1;
+    currentState = STATE_DICE_ROLL;
+    output += `<br> It is now Player ${currentPlayer}'s turn.`;
+  } else {
+    // get result of battle;
+    output += `<br>` + calculateWinner();
+
+    //SCORE - Adding Leadership Board
+    output += `<br><br> 
+    <h4> Leadership Board </h4>
+    Player 1: ${playerOneScore}
+    <br>
+    Player 2: ${playerTwoScore}`;
+  }
+
+  return output;
+};
+
+var updateNumberOfDiceInGame = function (diceNumber) {
+  if (diceNumber > NUMBER_OF_DICE_PLAYED) {
+    NUMBER_OF_DICE_PLAYED = diceNumber;
+    return diceNumber;
+  }
+  return NUMBER_OF_DICE_PLAYED;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  //hidden keywords trigger: MODE_LOWEST_WINS, AUTO_GENERATE_COMBINED_NUMBER
+  var myOutputValue = "";
+
+  //check if user changed game mode
+  if (input == MODE_LOWEST_WINS) {
+    currentMode = MODE_LOWEST_WINS;
+  }
+
+  if (currentState == STATE_DICE_ROLL) {
+    myOutputValue = getPlayerDisplayRoll();
+    currentState = STATE_DICE_ORDER;
+  } else {
+    //STATE_DICE_ORDER
+
+    ////VARIABLE NUMBER OF DICE - auto generate number if dice is more than 2
+    if (NUMBER_OF_DICE_PLAYED > 2)
+      currentState = STATE_AUTO_GENERATE_COMBINED_NUMBER;
+
+    if (input == STATE_AUTO_GENERATE_COMBINED_NUMBER) {
+      //updated to allow auto-generation of combined number
+      currentState = STATE_AUTO_GENERATE_COMBINED_NUMBER;
+      myOutputValue = getPlayerCombinedResult(input);
+    } else {
+      if (input < 0 || input > NUMBER_OF_DICE_PLAYED) {
```

Comment:
This could be refactored into a function. Just reading this logic does not really convey what's happening here